### PR TITLE
Update to SymEngine 0.12.0

### PR DESCRIPTION
**Pull request by anutosh491**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by anutosh491** [comment]
@conda-forge-admin, please rerender

**Comment by anutosh491** [comment]
cc @certik @isuruf might need some help from y'all addressing this.
So 
1) In the first commit I just changed the version and sha256 hash
2) Followed by the re-rendering step
3) Followed by addressing the suggestions by the linter.

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -1,2 +0,0 @@
-MACOSX_SDK_VERSION:    # [osx and x86_64]
-  - 10.12              # [osx and x86_64]
```
Comment (anutosh491) [inline recipe/conda_build_config.yaml:2]: Ohh so we won't need this ? I thought we had to match the version for `MACOSX_SDK_VERSION` and `c_stdlib_version` !

**Comment by anutosh491** [comment]
@conda-forge-admin, please rerender

**Comment by anutosh491** [comment]
@conda-forge-admin, please rerender

**Comment by anutosh491** [comment]
The build fails with the log 
```

    sys.exit(execute())
  File "/opt/conda/lib/python3.10/site-packages/conda_build/cli/main_build.py", line 590, in execute
    api.build(
  File "/opt/conda/lib/python3.10/site-packages/conda_build/api.py", line 250, in build
    return build_tree(
  File "/opt/conda/lib/python3.10/site-packages/conda_build/build.py", line 3638, in build_tree
    packages_from_this = build(
  File "/opt/conda/lib/python3.10/site-packages/conda_build/build.py", line 2419, in build
    try_download(m, no_download_source=False, raise_error=True)
  File "/opt/conda/lib/python3.10/site-packages/conda_build/render.py", line 762, in try_download
    source.provide(metadata)
  File "/opt/conda/lib/python3.10/site-packages/conda_build/source.py", line 1035, in provide
    unpack(
  File "/opt/conda/lib/python3.10/site-packages/conda_build/source.py", line 169, in unpack
    src_path, unhashed_fn = download_to_cache(
  File "/opt/conda/lib/python3.10/site-packages/conda_build/source.py", line 116, in download_to_cache
    raise RuntimeError(f"Could not download {url}")
RuntimeError: Could not download https://github.com/symengine/symengine/releases/download/v0.12.0/symengine-0.12.0.tar.gz
```

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (isuruf) [reply recipe/conda_build_config.yaml:2]: No need to set this. Please remove

**Comment by certik** [comment]
@isuruf I just tagged the SymEngine release. Do we need to manually upload a tarball also?

**Comment by isuruf** [comment]
Yes, I download the .tar.gz that github provides and reupload as an artifact. This is because the .tar.gz assets might change (when git protocol changes), but artifacts don't.

**Comment by certik** [comment]
I see. If one of you could do it, that would be awesome. I'll be able to do it on Monday the latest.

**Comment by anutosh491** [comment]
Hey @isuruf would you be able to look into the above comments. Not sure I can do those things as I lack member access to symengine.

**Comment by certik** [comment]
I downloaded the tarball and uploaded it explicitly as an asset: https://github.com/symengine/symengine/releases/tag/v0.12.0. 

@anutosh491 go ahead and try to get this PR working. I tried to restart the jobs, but it didn't do it. You probably have to push into this PR to get the CI running again.

**Comment by anutosh491** [comment]
@conda-forge-admin, please rerender

**Comment by anutosh491** [comment]
Hmm, not sure why the `symengine-feedstock (win win_64_)` would fail here returning an exception while testing `test_cwrapper`. All the windows 64 based test in the symengine CI pass perfectly on symengine's master !

**Comment by certik** [comment]
Well, this is a bug in SymEngine that was only discovered now. 

@isuruf we have two options:

* we uncomment running the test here and publish the conda packages; will fix this in symengine and make a new relesae
* we try to fix this in symengine, figure out a patch, and patch this release here

Which one would you prefer?

**Comment by isuruf** [comment]
I'd prefer the second option. I'll try to reproduce this locally.

**Comment by anutosh491** [comment]
Hey @isuruf were you able to repoduce the issue here ? Would be great if you could help us fix this. I'm presenting my work on symbolics with LPython at university in a weeks time and we would want to use symengine 0.12.0 there. So would be really helpful if you help us with this.

**Comment by anutosh491** [comment]
@conda-forge-admin, please rerender

**Comment by certik** [comment]
Let's get all tests passing, by disabling tests if needed, so that at least all conda packages build. We'll then decide how to go forward. We can also just disable the Windows build (since it doesn't build) and publish the linux/mac packages. That might be the preferable solution. But I think disabling the ctest is fine too.

**Comment by anutosh491** [comment]
Yes the test_cwrapper fails on win not sure why (everything looked good on symengine's master) and even after sending the patch for commenting that out the windows build fails. As you said we could disable the windows build for now, publish the mac/linux packages and look into the error later

**Comment by anutosh491** [comment]
@conda-forge-admin, please rerender

**Comment by anutosh491** [comment]
@conda-forge-admin, please rerender

**Comment by anutosh491** [comment]
@conda-forge-admin, rerender

**Comment by certik** [comment]
@isuruf, we have a path forward by disabling the Windows build for now. Are you ok with going forward like this, and once we figure out what the problem is, we fix it and re-enable Windows?

**Comment by isuruf** [comment]
Sure, that's fine for now

**Comment by certik** [comment]
Ok, I merged this. @anutosh491 is responsible to get the Windows build working again. I can help.

**Comment by anutosh491** [comment]
Yeah, let me raise an issue just to make a note of enabling the windows build again !

**Pull request merged**